x/ibc/core/02-client/keeper/client.go

```diff
@@ -95,13 +95,20 @@ func (k Keeper) UpdateClient(ctx sdk.Context, clientID string, header exported.H
 		)
 	}()
 
+	// emit the full header in events
+	var headerStr string
+	if header != nil {
+		headerStr = string(types.MustMarshalHeader(k.cdc, header))
```

Comment:
This might look really weird in an event log. Should we hex encode, rather than just converting straight to string?